tests/30rooms/60anonymousaccess.pl

```diff
@@ -482,6 +483,69 @@ sub await_event_not_presence_for
       });
    };
 
+test "Annonymous user calling /events doesn't tightloop",
+   requires => [ anonymous_user_fixture(), local_user_fixture() ],
+
+   do => sub {
+      my ( $anonymous_user, $user ) = @_;
+
+      my ( $room_id );
+
+      matrix_create_and_join_room( [ $user ] )
+      ->then( sub {
+         ( $room_id ) = @_;
+
+         matrix_set_room_history_visibility( $user, $room_id, "world_readable" );
+      })->then( sub {
+         do_request_json_for( $anonymous_user,
+            method => "GET",
+            uri    => "/api/v1/rooms/$room_id/initialSync",
+         );
+      })->then( sub {
+         my ( $sync_body ) = @_;
+         my $sync_from = $sync_body->{messages}->{end};
+
+         repeat( sub {
+            my ( undef, $f ) = @_;
+
+            my $body = $f ? $f->get : undef;
+            my $end_token = $body ? $body->{end} : $sync_from;
```

Comment:
You could just squash these into

    my $end_token = $f ? $f->get->{end} : $sync_from;